### PR TITLE
[MBL-18122][Parent] Create FAB Message Compose E2E Tests

### PR DESCRIPTION
**Pull request by kristofnemere**
Test plan: test event floating action button

refs: MBL-18122
affects: Parent
release note: none

## Checklist

- [x] Run E2E test suite

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

File: apps/parent/src/androidTest/java/com/instructure/parentapp/ui/e2e/InboxE2ETest.kt [hunk 1]
```diff
@@ -401,4 +409,133 @@ class InboxE2ETest: ParentComposeTest() {
...
         inboxPage.assertConversationNotDisplayed(seededConversation.subject)
     }
-}
\ No newline at end of file
+
+    @E2E
+    @Test
+    @TestMetaData(Priority.MANDATORY, FeatureCategory.INBOX, TestCategory.E2E)
+    fun testComposeMessageShortcutsE2E() {
+        Log.d(PREPARATION_TAG, "Seeding data.")
```
Comment (kdeakinstructure) [inline apps/parent/src/androidTest/java/com/instructure/parentapp/ui/e2e/InboxE2ETest.kt:417]: Please put an empty line under the method name.

File: apps/parent/src/androidTest/java/com/instructure/parentapp/ui/e2e/InboxE2ETest.kt [hunk 2]
```diff
@@ -401,4 +409,133 @@ class InboxE2ETest: ParentComposeTest() {
...
+        val data = seedData(students = 1, teachers = 1, parents = 1, courses = 2, syllabusBody = "Syllabus body")
+        val course = data.coursesList[0]
+        val courseWithFrontPage = data.coursesList[1]
+        val parent = data.parentsList[0]
+        val teacher = data.teachersList[0]
+        val student = data.studentsList[0]
+        PagesApi.createCoursePage(courseWithFrontPage.id, teacher.token, frontPage = true, editingRoles = "public")
```
Comment (kdeakinstructure) [inline apps/parent/src/androidTest/java/com/instructure/parentapp/ui/e2e/InboxE2ETest.kt:424]: A PREPARATION_TAG log would be nice here to log what we are seeding with API.

File: apps/parent/src/androidTest/java/com/instructure/parentapp/ui/e2e/InboxE2ETest.kt [hunk 3]
```diff
@@ -401,4 +409,133 @@ class InboxE2ETest: ParentComposeTest() {
...
+            }
+        ) {
+            coursesPage.clickCourseItem(course.name)
+        }
+        composeTestRule.waitForIdle()
+        courseDetailsPage.assertCourseNameDisplayed(course)
+
+        Log.d(STEP_TAG, "Click assignment ${testAssignment.name}.")
+        courseDetailsPage.clickAssignment(testAssignment.name)
+        composeTestRule.waitForIdle()
```
Comment (kdeakinstructure) [inline apps/parent/src/androidTest/java/com/instructure/parentapp/ui/e2e/InboxE2ETest.kt:456]: Does these composeTestRule.waitForIdle() really need after every action? I think it would work without these as well.

File: apps/parent/src/androidTest/java/com/instructure/parentapp/ui/e2e/InboxE2ETest.kt [hunk 4]
```diff
@@ -401,4 +409,133 @@ class InboxE2ETest: ParentComposeTest() {
...
+        courseDetailsPage.clickComposeMessageFAB()
+        composeTestRule.waitForIdle()
+        inboxComposeMessagePage.assertSubjectText(expectedSubjectFrontPage)
+
+        Log.d(STEP_TAG, "Go to Calendar and click on ${testEvent.title} event.")
+        inboxComposeMessagePage.pressBackButton()
+        inboxComposeMessagePage.pressExitConfirmationButton()
+        Espresso.pressBack()
+        dashboardPage.clickCalendarBottomMenu()
+        calendarScreenPage.clickOnItem(testEvent.title.orEmpty())
```
Comment (kdeakinstructure) [inline apps/parent/src/androidTest/java/com/instructure/parentapp/ui/e2e/InboxE2ETest.kt:519]: Do we need orEmpty() here? I don't think a seeded event with "Test Event" title can be empty here.

File: apps/parent/src/main/java/com/instructure/parentapp/features/calendarevent/ParentEventViewModelBehavior.kt [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+                subject = resources.getString(
+                    R.string.regardingHiddenMessageWithEventPrefix,
+                    parentPrefs.currentStudent?.name.orEmpty(),
+                    event.title.orEmpty()
+                )
+            ),
+            disabledFields = options.disabledFields.copy(
+                isContextDisabled = true
+            ),
+            autoSelectRecipientsFromRoles = listOf(EnrollmentType.TEACHERENROLLMENT),
```
Comment (kdeakinstructure) [inline apps/parent/src/main/java/com/instructure/parentapp/features/calendarevent/ParentEventViewModelBehavior.kt:54]: EnrollmentType value may could be snake case (TEACHER_ENROLLMENT)

File: automation/dataseedingapi/src/main/kotlin/com/instructure/dataseeding/api/CalendarEventApi.kt [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import com.instructure.dataseeding.model.ScheduleItemApiModel
+import com.instructure.dataseeding.util.CanvasNetworkAdapter
+import retrofit2.Call
+import retrofit2.http.POST
+import retrofit2.http.Query
+
+object CalendarEventApi {
+
+    interface CalendarEventService {
+        @POST("calendar_events/")
```
Comment (kdeakinstructure) [inline automation/dataseedingapi/src/main/kotlin/com/instructure/dataseeding/api/CalendarEventApi.kt:29]: I think it would work without the / character at the end

**Review by kdeakinstructure — APPROVED** [review]

File: apps/parent/src/main/java/com/instructure/parentapp/features/calendarevent/ParentEventViewModelBehavior.kt (hunk 5 above)
Comment (kristofnemere) [reply apps/parent/src/main/java/com/instructure/parentapp/features/calendarevent/ParentEventViewModelBehavior.kt:54]: it's in the legacy code, I won't rename it in this story
com.instructure.canvasapi2.type.EnrollmentType

File: apps/parent/src/androidTest/java/com/instructure/parentapp/ui/e2e/InboxE2ETest.kt (hunk 4 above)
Comment (kristofnemere) [reply apps/parent/src/androidTest/java/com/instructure/parentapp/ui/e2e/InboxE2ETest.kt:519]: if not orEmpty() then !!, I like this better

File: automation/dataseedingapi/src/main/kotlin/com/instructure/dataseeding/api/CalendarEventApi.kt (hunk 6 above)
Comment (kristofnemere) [reply automation/dataseedingapi/src/main/kotlin/com/instructure/dataseeding/api/CalendarEventApi.kt:29]: Copied from com.instructure.canvasapi2.apis.CalendarEventAPI.CalendarEventInterface#createCalendarEvent, but it works without it I guess

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — APPROVED** [review]
QA + 1

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

**Comment by inst-danger** [comment]
<p>Native Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>
